### PR TITLE
Private cached trie state

### PR DESCRIPTION
**Pull request by baptiste-b-pegasys**
This PR enables the same mechanism of cached and journalized trie state for the private state.

The default private journal is in the "privatetriestate" folder, configurable with `private.cache.trie.journal` flag, and must be different than the public journal (that is "triestate", configurable with `cache.trie.journal`)

For the other parameters, there are the same as the public cache.

With default parameters, there is an improvement of the TPS.

### CODE REVIEW

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -1699,12 +1706,19 @@ func setRaft(ctx *cli.Context, cfg *eth.Config) {
...
 	cfg.EVMCallTimeOut = time.Duration(ctx.GlobalInt(EVMCallTimeOutFlag.Name)) * time.Second
 	cfg.EnableMultitenancy = ctx.GlobalBool(MultitenancyFlag.Name)
 	cfg.SaveRevertReason = ctx.GlobalBool(RevertReasonFlag.Name)
 	setIstanbul(ctx, cfg)
 	setRaft(ctx, cfg)
+	if ctx.GlobalIsSet(PrivateCacheTrieJournalFlag.Name) {
+		cfg.PrivateTrieCleanCacheJournal = ctx.GlobalString(PrivateCacheTrieJournalFlag.Name)
+	}
+	if ctx.GlobalString(CacheTrieJournalFlag.Name) == cfg.PrivateTrieCleanCacheJournal {
+		return fmt.Errorf("configuation collision with '%s' and '%s' that must be different", CacheTrieJournalFlag.Name, PrivateCacheTrieJournalFlag.Name)
```
Comment (ricardolyn) [inline cmd/utils/flags.go]: typo `configuration`

File: core/mps/mock_interface.go [hunk 2]
```diff
@@ -37,34 +38,37 @@ func (m *MockPrivateStateManager) EXPECT() *MockPrivateStateManagerMockRecorder
 	return m.recorder
 }
 
-// ResolveForManagedParty mocks base method.
-func (m *MockPrivateStateManager) ResolveForManagedParty(managedParty string) (*PrivateStateMetadata, error) {
+// CheckAt mocks base method.
```
Comment (ricardolyn) [inline core/mps/mock_interface.go:41]: did we move the order of functions on this file?

Comment (baptiste-b-pegasys) [reply core/mps/mock_interface.go:41]: It's a generated file with golang mockgen, it might be a newer version

**Comment by ricardolyn** [comment]
@baptiste-b-pegasys there is a test that is failing. could you check?

**Review by ricardolyn — APPROVED** [review]
lgtm

**Review by nmvalera — APPROVED** [review]
LGTM

**Pull request merged**